internal/common/foraging/deerhunt.go

```diff
@@ -81,5 +86,5 @@ func (d DeerHunt) getPopulationLinkedProbability(dhConf config.DeerHuntConfig, p
 
 // Logf is a this type's custom logger
 func (d DeerHunt) Logf(format string, a ...interface{}) {
-	log.Printf("[SERVER][DeerHunt]: %v", fmt.Sprintf(format, a...))
+	log.Printf("[SERVER][DEERHUNT]: %v", fmt.Sprintf(format, a...))
```

Comment:
ah, this is fine, but i was actually thinking that when the server first initialises deerhunt, it passes the pointer to the server's logf function (s.Logf). d.Logf would then call the logger with just `[DEERHUNT]` preprended.

Reply:
so `NewDeerHunt(s.Logf, ...)`

```
type DeerHunt struct {
     logger func(format string, ...)
}
func NewDeerHunt(logger, ...) {
    return DeerHunt{ logger: logger, ... }
}
func (d *DeerHunt) Logf(format, ...) {
    d.logger.Logf("[DEERHUNT]: ...")
}
```

Reply:
separate PR encouraged :) you can also show how it should be done for  iigointernal, I think they're struggling....

Reply:
Okay cool. Will do in a separate PR. Let me know if anything else to add in this one pls :) @lhl2617

Reply:
it's good let's goooooo